javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -1044,9 +1044,9 @@ protected JavaVisibility methodVisibility(
                     : NOT_GENERATE;
             }
         } else {
-            if (!settings.isSyncStackEnabled() && methodType == ClientMethodType.SimpleSyncRestResponse && !hasContextParameter) {
+            if (methodType == ClientMethodType.SimpleSyncRestResponse && !hasContextParameter) {
                 return NOT_GENERATE;
-            } else if (!settings.isSyncStackEnabled() && methodType == ClientMethodType.SimpleSync && hasContextParameter) {
+            } else if (methodType == ClientMethodType.SimpleSync && hasContextParameter) {
                 return NOT_GENERATE;
             }
```

Comment:
nit, the 2 methods might be needed for internal call within client. You can regen the test and see if some error happens. If they are needed, one could use NOT_VISIBLE.